src/entities/selectionBox.ts

```diff
@@ -0,0 +1,125 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Entity {
+  export class SelectionBox {
+    private _parentNode: Element;
+    private _box: D3.Selection;
+    private _core: D3.Selection;
+    private _edgeT: D3.Selection;
+    private _edgeB: D3.Selection;
+    private _edgeL: D3.Selection;
+    private _edgeR: D3.Selection;
+    private _cornerTL: D3.Selection;
+    private _cornerTR: D3.Selection;
+    private _cornerBL: D3.Selection;
+    private _cornerBR: D3.Selection;
+
+    private _topLeft: Point = { x: 0, y: 0 };
+    private _bottomRight: Point = { x: 0, y: 0 };
+
+    private _detectionRadius = 2;
```

Comment:
Going as much as we can about decoupling interactions and visual items, is an idea of a _detectionRadius digging a bit far into the realm of interaction logic?  I'm thinking a rename here is all we need since this seems to be determining the stroke widths of the lines?